Fill.py

```diff
@@ -106,8 +106,21 @@ def fill_restrictive(world: MultiWorld, base_state: CollectionState, locations,
 def distribute_items_restrictive(world: MultiWorld, fill_locations=None):
     # If not passed in, then get a shuffled list of locations to fill in
     if not fill_locations:
-        fill_locations = world.get_unfilled_locations()
-        world.random.shuffle(fill_locations)
+        fill_locations = world.get_locations()
+
+    world.random.shuffle(fill_locations)
+
+    prioritizedlocations = []
```

Comment:
instead of 3 lists it might be "easier" to make a {progress_type: [locations]} dict and fill it with locations[location.progress_type].append(location)

Reply:
Ordered using a dict, but I create the lists after for easier access and reference.